pkg/endpointmanager/manager.go

```diff
@@ -210,8 +210,9 @@ func Remove(ep *endpoint.Endpoint) {
 		// While restoring, endpoint IDs may not have been reused yet.
 		// Failure to release means that the endpoint ID was not reused
 		// yet. Avoid irritating warning messages.
-		if ep.GetStateLocked() != endpoint.StateRestoring {
-			log.WithError(err).Warning("Unable to release endpoint ID")
+		state := ep.GetStateLocked()
+		if state != endpoint.StateRestoring && state != endpoint.StateDisconnecting {
```

Comment:
An update to the comment / commit message would be helpful here. Why is it OK that we are silencing errors around releasing endpoint IDs that are in "disconnecting" state? Is this because the ID has already been released when it is in this state? I think that it's important to justify why this is the right fix, as it is not apparent to me.

Reply:
Agreed, please add a comment. It's not obvious to me why disconnecting has to be in that list.

Reply:
Add that same comment to the commit message as well

Reply:
Based on my tests, this happened with endpoint being in `disconnecting` state, while endpoint ID was already available in id cache, which caused `pool.Insert` in `pkg/endpoint/id/allocator.go` call to return false, causing the error, resulting in (IMO) unnecessary warning.